### PR TITLE
3261/adding challenge_type to multi_challenge

### PR DESCRIPTION
**Pull request by jona-samuel**
This adds a new entry to the reply_dict of the multi_challenge with information about the challenge type.
More info in #3261

### CODE REVIEW

**Review by lukasmatusiewicz — APPROVED** [review]

File: privacyidea/lib/challengeresponsedecorators.py [hunk 1]
```diff
@@ -59,9 +59,14 @@ def _create_challenge(token_obj, challenge_type, message, challenge_data=None):
     db_challenge.save()
     token_obj.challenge_janitor()
     reply_dict = {}
+    if ((challenge_type == CHALLENGE_TYPE.PIN_RESET) or (challenge_type == CHALLENGE_TYPE.RESYNC)):
+        challenge = challenge_type
+    else:
+        challenge = "standard"
```
Comment (plettich) [inline privacyidea/lib/challengeresponsedecorators.py:65]: This should be added to the [`CHALLENGE_TYPE` class](https://github.com/privacyidea/privacyidea/blob/28f138522618915950769a0fe196720f3b417393/privacyidea/lib/challengeresponsedecorators.py#L46), maybe as `generic_challenge` or `standard_challenge` (@cornelinux should know, why the values are called `generic_*`).

File: tests/test_lib_token.py [hunk 2]
```diff
@@ -1934,6 +1934,7 @@ def test_01_successfully_change_pin(self):
                                                   "g": g})
         self.assertFalse(r)
         self.assertEqual("Please enter a new PIN", reply_dict.get("message"))
+        self.assertEqual("generic_pin_reset", reply_dict.get('multi_challenge')[0].get('challenge_type'))
```
Comment (plettich) [inline tests/test_lib_token.py:1937]: We should also test somewhere, that the `standard` challenge type is returned if there is no PIN-reset or resync-challenge.

**Review by plettich — CHANGES_REQUESTED** [review]

File: privacyidea/lib/challengeresponsedecorators.py (hunk 1 above)
Comment (jona-samuel) [reply privacyidea/lib/challengeresponsedecorators.py:65]: @lukasmatusiewicz is 'generic_challenge' instead of 'standard' fine for you?

Comment (jona-samuel) [reply privacyidea/lib/challengeresponsedecorators.py]: 7d835b180677b94d0de6d320c9d8841caff697a1

File: tests/test_lib_token.py (hunk 2 above)
Comment (jona-samuel) [reply tests/test_lib_token.py:1939]: 7d835b180677b94d0de6d320c9d8841caff697a1

File: privacyidea/lib/token.py [hunk 3]
```diff
@@ -2092,6 +2092,7 @@ def create_challenges_from_tokens(token_list, reply_dict, options=None):
             message_list.append(message)
             if r_chal:
                 challenge_info = challenge_info or {}
+                challenge_info["challenge_type"] = "generic_challenge"
```
Comment (plettich) [inline privacyidea/lib/token.py:2095]: please add the `generic_challenge` to the `CHALLENGE_TYPE` class here:
https://github.com/privacyidea/privacyidea/blob/28f138522618915950769a0fe196720f3b417393/privacyidea/lib/challengeresponsedecorators.py#L46

**Review by plettich — CHANGES_REQUESTED** [review]

**Comment by jona-samuel** [comment]
this is replaced with #3569

**Pull request closed without merging**